engine/src/lib.rs

```diff
@@ -1,20 +1,40 @@
-pub mod board;
-pub mod bug;
-pub mod bug_stack;
-pub mod color;
-pub mod dfs_info;
-pub mod direction;
-pub mod game_control;
-pub mod game_error;
-pub mod game_result;
-pub mod game_status;
-pub mod game_type;
-pub mod hasher;
-pub mod history;
-pub mod mid_move_board;
-pub mod piece;
-pub mod player;
-pub mod position;
-pub mod state;
-pub mod torus_array;
-pub mod turn;
+mod board;
+mod bug;
+mod bug_stack;
+mod color;
+mod dfs_info;
+mod direction;
+mod game_control;
+mod game_error;
+mod game_result;
+mod game_status;
+mod game_type;
+mod hasher;
+mod history;
+mod mid_move_board;
+mod piece;
+mod player;
+mod position;
+mod state;
+mod torus_array;
+mod turn;
+pub use board::Board;
+pub use bug::Bug;
+pub use bug_stack::BugStack;
+pub use color::{Color, ColorChoice};
+pub use dfs_info::DfsInfo;
```

Comment:
`DfsInfo` and `MidMoveBoard` are internal types and it should not be accessible from the outside.
Unsure about the `TorusArray`...

Reply:
You are correct, none of these types are required to be public, the Hasher type as well